Lib/test/test_httpservers.py

```diff
@@ -315,6 +331,30 @@ def test_head_via_send_error(self):
             self.assertEqual(b'', data)
 
 
+@unittest.skipIf(ssl is None, 'No ssl module')
+class BaseHTTPSServerTestCase(BaseTestCase):
+    tls = (
+        os.path.join(os.path.dirname(__file__), "certdata", "ssl_cert.pem"),
+        os.path.join(os.path.dirname(__file__), "certdata", "ssl_key.pem"),
+    )
+
+    class request_handler(NoLogRequestHandler, SimpleHTTPRequestHandler):
+        pass
+
+    def test_get(self):
```

Comment:
Can we have more test where an incorrect certificate chain is provided, or an invalid key, or an invalid password as well? There is a lack of coverage for the `password=...` support. In addition, it could be good to test that we can't instantiate the HTTPS server when SSL is missing.

Reply:
I added tests :fireworks: